### ISSUE TITLE
dart2js failed to compile, crash with Internal Error: Cannot find value BoxLocal(_box_0)

### ISSUE THREAD

**Issue by avenwu** [dart-lang/sdk#43620]
When I compiled one of my flutter-web project, the complier crashed and made a prompt to fire bug here.
 
This crash only happens when I tried to build from cli. It works well when I launch the project from AndroidStudio in debug mode, a chrome tab would be opened automatically and the display the html correctly.
## Command used
```shell
flutter build web
```
## System information
* macOS Catalina Version 10.15.5
*  Dart SDK build number (2.9.2)
* Flutter SDK 1.20.4

>  flutter doctor

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel unknown, 1.20.4, on Mac OS X 10.15.5 19F101, locale en-CN)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 11.3)
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.0)
[✓] VS Code (version 1.30.0)
[✓] Connected device (3 available)

• No issues found!
```

## Crash log
```
Running "flutter pub get" in example...                             0.5s
Target dart2js failed: Exception: ../../../../Android/flutter/packages/flutter/lib/src/widgets/widget_inspector.dart:2517:3:
Internal Error: Cannot find value BoxLocal(_box_0) in (SyntheticLocal(result), local(_InspectorOverlayLayer.#overlayRect),
local(_InspectorOverlayLayer.#selection)) for j:constructor(_InspectorOverlayLayer.).
  _InspectorOverlayLayer({
  ^
The compiler is broken.

When compiling the above element, the compiler crashed. It is not
possible to tell if this is caused by a problem in your program or
not. Regardless, the compiler should not crash.

The Dart team would greatly appreciate if you would take a moment to
report this problem at http://dartbug.com/new.

Please include the following information:

* the name and version of your operating system,

* the Dart SDK build number (2.9.2), and

* the entire message you see here (including the full stack trace
  below as well as the source location above).

Unhandled exception:
Assertion failure: Cannot find value BoxLocal(_box_0) in (SyntheticLocal(result), local(_InspectorOverlayLayer.#overlayRect),
local(_InspectorOverlayLayer.#selection)) for j:constructor(_InspectorOverlayLayer.).
#0      failedAt (package:compiler/src/diagnostics/invariant.dart:49:3)
#1      LocalsHandler.readLocal (package:compiler/src/ssa/locals_handler.dart:331:11)
#2      KernelSsaGraphBuilder._buildConstructor.<anonymous closure> (package:compiler/src/ssa/builder_kernel.dart:862:44)
#3      KernelSsaGraphBuilder._inlinedFrom.<anonymous closure> (package:compiler/src/ssa/builder_kernel.dart:920:21)
#4      CompilerDiagnosticReporter.withCurrentElement (package:compiler/src/compiler.dart:743:15)
#5      KernelSsaGraphBuilder._inlinedFrom (package:compiler/src/ssa/builder_kernel.dart:918:14)
#6      KernelSsaGraphBuilder._buildConstructor (package:compiler/src/ssa/builder_kernel.dart:837:7)
#7      KernelSsaGraphBuilder._visitInlinedFunction (package:compiler/src/ssa/builder_kernel.dart:5997:9)
#8      KernelSsaGraphBuilder._doInline (package:compiler/src/ssa/builder_kernel.dart:5987:5)
#9      KernelSsaGraphBuilder._tryInlineMethod.doInlining.<anonymous closure> (package:compiler/src/ssa/builder_kernel.dart:5669:11)
#10     KernelSsaGraphBuilder._inlinedFrom.<anonymous closure> (package:compiler/src/ssa/builder_kernel.dart:920:21)
#11     CompilerDiagnosticReporter.withCurrentElement (package:compiler/src/compiler.dart:743:15)
#12     KernelSsaGraphBuilder._inlinedFrom (package:compiler/src/ssa/builder_kernel.dart:918:14)
#13     KernelSsaGraphBuilder._tryInlineMethod.doInlining (package:compiler/src/ssa/builder_kernel.dart:5664:7)
#14     KernelSsaGraphBuilder._tryInlineMethod (package:compiler/src/ssa/builder_kernel.dart:5676:17)
#15     KernelSsaGraphBuilder._pushStaticInvocation (package:compiler/src/ssa/builder_kernel.dart:4730:9)
#16     KernelSsaGraphBuilder.visitConstructorInvocation (package:compiler/src/ssa/builder_kernel.dart:5287:5)
#17     ConstructorInvocation.accept (package:kernel/ast.dart:4154:44)
#18     KernelSsaGraphBuilder._visitPositionalArguments (package:compiler/src/ssa/builder_kernel.dart:3540:16)
#19     KernelSsaGraphBuilder._visitArgumentsForDynamicTarget (package:compiler/src/ssa/builder_kernel.dart:3553:33)
#20     KernelSsaGraphBuilder.visitMethodInvocation (package:compiler/src/ssa/builder_kernel.dart:5038:42)
#21     MethodInvocation.accept (package:kernel/ast.dart:3950:44)
#22     KernelSsaGraphBuilder.visitExpressionStatement (package:compiler/src/ssa/builder_kernel.dart:1714:18)
#23     ExpressionStatement.accept (package:kernel/ast.dart:5784:43)
#24     KernelSsaGraphBuilder.visitBlock (package:compiler/src/ssa/builder_kernel.dart:1680:17)
#25     Block.accept (package:kernel/ast.dart:5823:43)
#26     KernelSsaGraphBuilder._buildFunctionNode (package:compiler/src/ssa/builder_kernel.dart:1275:23)
#27     KernelSsaGraphBuilder.build.<anonymous closure> (package:compiler/src/ssa/builder_kernel.dart:443:15)
#28     CompilerDiagnosticReporter.withCurrentElement (package:compiler/src/compiler.dart:743:15)
#29     KernelSsaGraphBuilder.build (package:compiler/src/ssa/builder_kernel.dart:428:21)
#30     KernelSsaBuilder.build.<anonymous closure> (package:compiler/src/js_model/js_strategy.dart:500:22)
#31     CompilerTask.measure (package:compiler/src/common/tasks.dart:64:51)
#32     KernelSsaBuilder.build (package:compiler/src/js_model/js_strategy.dart:483:18)
#33     SsaBuilderTask.build (package:compiler/src/ssa/ssa.dart:346:21)
#34     SsaFunctionCompiler.compile (package:compiler/src/ssa/ssa.dart:80:29)
#35     OnDemandCodegenResults.getCodegenResults (package:compiler/src/common/codegen.dart:454:30)
#36     JsBackendStrategy.generateCode (package:compiler/src/js_model/js_strategy.dart:319:43)
#37     KernelCodegenWorkItem.run (package:compiler/src/js_model/js_strategy.dart:450:29)
#38     Compiler.emptyQueue.<anonymous closure>.<anonymous closure>.<anonymous closure>.<anonymous closure>.<anonymous closure>
(package:compiler/src/compiler.dart:460:70)
#39     CompilerTask.measureSubtask (package:compiler/src/common/tasks.dart:179:35)
#40     Compiler.emptyQueue.<anonymous closure>.<anonymous closure>.<anonymous closure>.<anonymous closure>
(package:compiler/src/compiler.dart:460:32)
#41     CompilerTask.measureSubtask (package:compiler/src/common/tasks.dart:179:35)
#42     Compiler.emptyQueue.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:compiler/src/compiler.dart:458:28)
#43     CompilerDiagnosticReporter.withCurrentElement (package:compiler/src/compiler.dart:743:15)
#44     Compiler.emptyQueue.<anonymous closure>.<anonymous closure> (package:compiler/src/compiler.dart:456:18)
#45     CodegenEnqueuer._forEach (package:compiler/src/js_backend/enqueuer.dart:266:12)
#46     CodegenEnqueuer.forEach (package:compiler/src/js_backend/enqueuer.dart:282:5)
#47     Compiler.emptyQueue.<anonymous closure> (package:compiler/src/compiler.dart:452:16)
#48     CompilerTask.measureSubtask (package:compiler/src/common/tasks.dart:179:35)
#49     Compiler.emptyQueue (package:compiler/src/compiler.dart:451:14)
#50     Compiler.processQueue.<anonymous closure> (package:compiler/src/compiler.dart:477:7)
#51     CompilerTask.measureSubtask (package:compiler/src/common/tasks.dart:179:35)
#52     Compiler.processQueue (package:compiler/src/compiler.dart:470:14)
#53     Compiler.runCodegenEnqueuer (package:compiler/src/compiler.dart:382:5)
#54     Compiler.generateJavaScriptCode (package:compiler/src/compiler.dart:284:9)
#55     Compiler.compileFromKernel.<anonymous closure> (package:compiler/src/compiler.dart:432:9)
#56     CompilerTask.measureSubtask (package:compiler/src/common/tasks.dart:179:35)
#57     Compiler.compileFromKernel (package:compiler/src/compiler.dart:405:14)
#58     Compiler.runInternal (package:compiler/src/compiler.dart:257:13)
<asynchronous suspension>
#59     Compiler.run.<anonymous closure>.<anonymous closure> (package:compiler/src/compiler.dart:212:38)
#60     new Future.sync (dart:async/future.dart:223:31)
#61     Compiler.run.<anonymous closure> (package:compiler/src/compiler.dart:212:20)
#62     CompilerTask.measureSubtask (package:compiler/src/common/tasks.dart:179:35)
#63     Compiler.run (package:compiler/src/compiler.dart:209:41)
#64     CompilerImpl.run.<anonymous closure>.<anonymous closure> (package:compiler/src/apiimpl.dart:91:22)
#65     _RootZone.runUnary (dart:async/zone.dart:1450:54)
#66     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
#67     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
#68     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
#69     Future._completeWithValue (dart:async/future_impl.dart:529:5)
#70     Future._asyncCompleteWithValue.<anonymous closure> (dart:async/future_impl.dart:567:7)
#71     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#72     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#73     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
#74     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:169:5)

```

**Comment by avenwu** [comment]
After reading source code relative to the crash log. I'm able to skip this compile crash by remove any reference of `WidgetInspector`
> flutter/packages/flutter/lib/src/widgets/widget_inspector.dart:2517:3

Line 2517 of widget_inspector.dart define the constructor of `_InspectorOverlayLayer`, and it's referenced by WidgetInspector, I was not using this widget, so remove it works for me.
<img width="733" alt="Screen Shot 2020-10-10 at 7 34 51 PM" src="https://user-images.githubusercontent.com/1622234/95654070-07ac4c00-0b30-11eb-9c05-4404b28d716c.png">

**Comment by sigmundch** [comment]
Thanks for the update, glad that you are unblocked.

I forgot to ask, but by any chance is the issue fixed  the latest stable release (Dart 2.10.1)?

There is a possibility that the change @fishythefish made in 7e8348f could help here (which was released in 2.10.0 for the first time).

**Comment by avenwu** [comment]
I was not able to switch to dart 2.10.1 for my project currently, will try when possible.
